momepy/functional/_distribution.py

```diff
@@ -339,3 +344,124 @@ def street_alignment(
+        The ``np.array`` or ``pd.Series``
+        where object orientation values are stored. This
+        can be calculated using :func:`orientation`.
+
+    Returns
+    -------
+    Series
+    """
+    return (left_orientations - right_orientations).abs()
+
+
+def shared_walls_ratio(
+    shared_walls: np.ndarray | Series, perimeters: np.ndarray | Series
+):
+    """Calculate shared walls ratio of adjacent elements (typically buildings).
+
+    .. math::
+        \\textit{length of shared walls} \\over perimeter
+
+    Note that data needs to be topologically correct.
+    Overlapping polygons will lead to incorrect results.
+
+    Adapted from :cite:`hamaina2012a`.
+
+    Parameters
+    ----------
+    shared_walls : np.array, pd.Series
+        The  ``np.array``, or
+        `pd.Series`` shared wall polygons are stored. This
+        can be calculated using :func:`shared_walls`.
+    perimeters : np.array, pd.Series
+        The  ``np.array``, or
+        `pd.Series`` containing used perimeters values.
+        Typically the perimeters of the original GeoDataFrame
+        used to generate the shared walls.
+
+    Returns
+    -------
+    Series
+    """
+
+    return shared_walls / perimeters
+
+
+def _orient(geom):
+    start = geom.coords[0]
+    end = geom.coords[-1]
+    az = _azimuth(start, end)
+    if 90 > az >= 45:
+        diff = az - 45
+        az = az - 2 * diff
+    elif 135 > az >= 90:
+        diff = az - 90
+        az = az - 2 * diff
+        diff = az - 45
+        az = az - 2 * diff
+    elif 181 > az >= 135:
+        diff = az - 135
+        az = az - 2 * diff
+        diff = az - 90
+        az = az - 2 * diff
+        diff = az - 45
+        az = az - 2 * diff
+    return az
+
+
+def neighboring_street_orientation_deviation(gdf: gpd.GeoDataFrame) -> Series:
+    """Calculate the mean deviation of solar orientation of adjacent streets.
+    The orientation of a street segment is represented by the orientation of the line
+    connecting the first and last point of the segment.
+
+    .. math::
+        \\frac{1}{n}\\sum_{i=1}^n dev_i=\\frac{dev_1+dev_2+\\cdots+dev_n}{n}
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+        A GeoDataFrame containing objects to analyse.
+
+    Returns
+    -------
+    Series
+    """
+
+    orientation = gdf.geometry.apply(_orient)
+
+    if GPD_GE_013:
+        inp, res = gdf.sindex.query(gdf.geometry, predicate="intersects")
+    else:
+        inp, res = gdf.sindex.query_bulk(gdf.geometry, predicate="intersects")
+    itself = inp == res
+    inp = inp[~itself]
+    res = res[~itself]
+
+    left = orientation.take(inp).reset_index(drop=True)
+    right = orientation.take(res).reset_index(drop=True)
+    deviations = (left - right).abs()
+    stats = deviations.groupby(inp).mean()
+
+    results = Series(np.nan, gdf.index.values)
+    # iloc since sindex query returns a numpy iloc array
+    results.iloc[stats.index.values] = stats.values
+    return results
```

Comment:
If we get a graph based on street edges, we can do this using `describe`. I'd say we point users there.